src/invidious.cr

```diff
@@ -66,7 +66,8 @@ SOFTWARE = {
   "branch"  => "#{CURRENT_BRANCH}",
 }
 
-YT_POOL = YoutubeConnectionPool.new(YT_URL, capacity: CONFIG.pool_size, timeout: 2.0, use_quic: CONFIG.use_quic)
+YT_POOL      = YoutubeConnectionPool.new(YT_URL, capacity: CONFIG.pool_size, timeout: 2.0, use_quic: CONFIG.use_quic)
+MAX_LOOKBACK = 20
```

Comment:
True, 20 might be too low. It's now configurable, but I put 50 as default.

Imo it shouldn't be too close to the total number of videos in a playlist, otherwise this will happen.
* max_lookback ->  90
* playlist size -> 1000
* Requesting playlist at video # 98 
* Looks back to 0
* Gets playlist 0 (...) 98 (...) 99

Or you can just increase the max number of returned videos. 100 lookback with 200 max_videos . Which in a big playlist means making 2 youtube requests per request. Up to you.

Reply:
I think that it would be better to stay at 1 API call (so 100 videos), and try to get up to the 49 preceeding / 50 following videos (all following videos if `idx < 50` / all following videos if `idx > end-50`) on the backend, then add javascript to request more as the user scrolls.